src/v/kafka/server/quota_manager.cc

```diff
@@ -32,12 +32,48 @@ quota_manager::quota_manager()
       config::shard_local_cfg().kafka_admin_topic_api_rate.bind())
   , _target_tp_rate_per_client_group(
       config::shard_local_cfg().kafka_client_group_byte_rate_quota.bind())
+  , _kafka_throughput_limit_node_in_bps(
+      config::shard_local_cfg().kafka_throughput_limit_node_in_bps.bind())
+  , _kafka_throughput_limit_node_out_bps(
+      config::shard_local_cfg().kafka_throughput_limit_node_out_bps.bind())
+  , _kafka_quota_balancer_window(
+      config::shard_local_cfg().kafka_quota_balancer_window.bind())
+  , _shard_ingress_quota(
+      _kafka_throughput_limit_node_in_bps() / ss::smp::count,
+      _kafka_quota_balancer_window())
```

Comment:
i assume this behaves well when core count is larger than bps?

Reply:
No it is not, there will be a vassert on `quota > 0`. That's why the config property minimum is set to `ss::smp::count`.

Reply:
i think you should handle that case here rather than relying on the config value being in a range related to ss::smp::count.

Reply:
I'd say keep the config property constraint and add additional error checking here as well. The additional error check is probably useful for when this class is used in another context like unit tests.

Reply:
I have decided to refactor these properties, following your comments and also considering that it's no good for a cluster property (that is persisted in the controller log) to have min/max constraints depend on an environmental parameter. A node can find itself in violation of these constraints if the number of cores is different after a restart (--smp, or VM, or migration to different hardware). I'm making these properties optional (to represent the unlimited option), and the constraints will be soft.